library/core/src/sync/atomic.rs

```diff
@@ -272,7 +272,7 @@ unsafe impl Sync for AtomicBool {}
 
 /// A raw pointer type which can be safely shared between threads.
 ///
-/// This type has the same in-memory representation as a `*mut T`.
+/// This type has the same size, alignment, and bit validity as a `*mut T`.
```

Comment:
```suggestion
/// This type has the same size and bit validity as a `*mut T`.
```

`*mut T` can have a smaller alignment than its size (e.g., on AVR, size is 16-bit and alignment is 8-bit).